homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect, useCallback } from "react";
+import UserCard from "../components/UserCard";
+import UserInfo from "../components/UserInfo/UserInfo";
+import Spinner from "../components/Spinner";
+import Button from "../components/Button";
+import "./HomeworkWeek3.css";
+
+function HomeWorkWeek3() {
+  const API = "https://uinames.com/api/?ext";
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      const res = await fetch(API);
+      const usersData = await res.json();
+      setIsLoading(false);
+      setData(usersData);
+    }
+    fetchData();
+
+    return function cleanup() {
+      return fetchData();
+    };
+  }, []);
+
+  const onHandleClick = useCallback(() => {
+    async function fetchNewUser() {
+      setIsLoading(true);
+      const res = await fetch(API);
+      const newUserData = await res.json();
+      setIsLoading(false);
+      setData(newUserData);
+    }
+    fetchNewUser();
+
+    return function cleanup() {
+      return fetchNewUser();
+    };
+  }, []);
```

Comment:
Cool that you used `useCallback`. This code implements almost the same as on line 13 to 26 though. Try to find a way that you can use either the `useEffect` in both cases or this `useCallback` implementation (to not repeat code unnecessary)